### ISSUE TITLE
[Bug]: `"TypeError: defaultSearchParams.keys() is not iterable"` in Firefox Addon content scripts

### ISSUE THREAD

**Issue by smithki** [remix-run/react-router#10618]
### What version of React Router are you using?

6.11.0

### Steps to Reproduce

There is a known issue in Firefox — still open after 9 years — that manifests in the following TypeError when using React Router DOM in the content scripts of a Firefox Addon:

<img width="513" alt="Screenshot 2023-06-16 at 4 26 29 PM" src="https://github.com/remix-run/react-router/assets/22753882/fe13ad4a-afe0-4fd7-846a-33012f80e931">

```
TypeError: defaultSearchParams.keys() is not iterable[dom.ts:100:40](moz-extension://ea565e18-3799-46da-aa12-608dc044d7e7/node_modules/react-router-dom/dom.ts)
    getSearchParamsForLocation dom.ts:100
```

Here's a link to the relevant Bugzilla ticket: https://bugzilla.mozilla.org/show_bug.cgi?id=1023984

### Expected Behavior

`useSearchParams` doesn't not produce an error.

### Actual Behavior

React experiences a fatal error at runtime, including recursive `"Invalid hook call"` warnings. This is a showstopper for anyone who wants to use React Router DOM in a web extension (for my case, paired with `MemoryRouter` to simplify routing for UIs that are injected into content scripts).

**Comment by smithki** [comment]
Though this is ultimately a shortcoming of Firefox, I've confirmed that a small change to [`getSearchParamsForLocation()`](https://github.com/remix-run/react-router/blob/7e2a122cc3b9739e560562bbb1772bdc86eb7f34/packages/react-router-dom/dom.ts#L93) should fix the bug and enable the web extension use-case.

```ts
// line 100 in packages/react-router-dom/dom.ts
for (let key of defaultSearchParams.keys()) {
```

should become:

```ts
defaultSearchParams.forEach((key) => { ...
```

**Comment by brophdawg11** [comment]
This is fixed by #10620.  It merged to `dev` after we started the release process for `6.14.2` (currently in prerelease), so it'll be available in the next release after that 👍

**Comment by brophdawg11** [comment]
We just published version `6.15.0-pre.0` which involves this issue. If you'd like to take it for a test run please try it out and let us know what you think!
